complete-application/src/index.ts

```diff
@@ -112,6 +115,62 @@ app.get("/account", async (req, res) => {
 });
 //end::account[]
 
+//tag::make-change[]
+app.get("/make-change", async (req, res) => {
+    const userTokenCookie = req.cookies[userToken];
+
+    // Make sure the user is authenticated. Note that in a production application, we would validate the token signature, 
+    // make sure it wasn't expired, and attempt to refresh it if it were
+    if (!userTokenCookie) {
+        res.redirect(302, '/');
+    } else {
+        res.sendFile(path.join(__dirname, '../templates/make-change.html'));
+    }
+});
+
+app.post("/make-change", async (req, res) => {
+    const userTokenCookie = req.cookies[userToken];
+
+    // Make sure the user is authenticated. Note that in a production application, we would validate the token signature, 
+    // make sure it wasn't expired, and attempt to refresh it if it were
+    if (!userTokenCookie) {
+        res.status(403).json(JSON.stringify({
+            error: 'Unauthorized'
+        }))
+    } else {
+
+        let error;
+        let message;
+
+        var coins = {
+            quarters: 0.25,
+            dimes: 0.1,
+            nickels: 0.05,
+            pennies: 0.01,
+        };
+
+        try {
+            message = 'We can make change for';
```

Comment:
There's something wrong with the rounding: If I put 26 cents, I get 1 quarter and 2 pennies. The same thing happens with 27, 28, 51, 54 cents, and so on.

I think you should multiply both `req.body.amount` and `coins` * 100 to get rid of decimals and just use `remainingAmount = remainingAmount % nominal;` to avoid rounding errors.

> I hate how most languages are terrible at this.